src/trix/elements/trix_editor_element.coffee

```diff
@@ -114,7 +114,12 @@ Trix.registerElement "trix-editor", do ->
         @trixId
 
   toolbarElement:
+    set: (value) ->
+      @toolbar = value
     get: ->
+      if @toolbar
+        return @toolbar
+
```

Comment:
This solution isn't ideal since it *requires* assigning the toolbar element for all editors in shadow roots. It also doesn't account for hidden input elements, which suffer the same problem there.

A better solution might involve [`Node.getRootNode()`](https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode):
> * Calling it on an element inside a standard web page will return an `HTMLDocument` object representing the entire page.
> * Calling it on an element inside a shadow DOM will return the associated `ShadowRoot`.

and might look something like:
```diff
--- a/src/trix/elements/trix_editor_element.coffee
+++ b/src/trix/elements/trix_editor_element.coffee
@@ -105,6 +105,10 @@ Trix.registerElement "trix-editor", do ->
 
   # Properties
 
+  rootNode:
+    get: ->
+      @getRootNode?() ? @ownerDocument
+
   trixId:
     get: ->
       if @hasAttribute("trix-id")
@@ -116,7 +120,7 @@ Trix.registerElement "trix-editor", do ->
   toolbarElement:
     get: ->
       if @hasAttribute("toolbar")
-        @ownerDocument?.getElementById(@getAttribute("toolbar"))
+        @rootNode?.getElementById(@getAttribute("toolbar"))
       else if @parentNode
         toolbarId = "trix-toolbar-#{@trixId}"
         @setAttribute("toolbar", toolbarId)
@@ -127,7 +131,7 @@ Trix.registerElement "trix-editor", do ->
   inputElement:
     get: ->
       if @hasAttribute("input")
-        @ownerDocument?.getElementById(@getAttribute("input"))
+        @rootNode?.getElementById(@getAttribute("input"))
       else if @parentNode
         inputId = "trix-input-#{@trixId}"
         @setAttribute("input", inputId)
```
Mind giving that ^ a try?

Reply:
@javan I tried out your code and found that it _would_ work, but unfortunately the framework I'm using disallows certain methods on the shadow root, including geElementById. Check this line out: https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine/src/framework/restrictions.ts#L285

Seems like the easiest way for me to set the toolbar is with a DOM reference

Reply:
Okay, I found that I could make it work by using querySelector instead of getElementById because LWC blocks usage of getElementById (which sucks, I know).